ifopt_core/CMakeLists.txt

```diff
@@ -34,35 +35,54 @@ add_library(${PROJECT_NAME}
 #############
 ## Install ##
 #############
-# Mark library for installation
-install(
-  TARGETS ${PROJECT_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
-
-# Mark header files for installation
-install(
-  DIRECTORY include/ifopt/
-  DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}/ifopt
-  FILES_MATCHING PATTERN "*.h"
-)
-
+if(${CATKIN_BUILD})
+    # Mark library for installation
+    install(
+      TARGETS ${PROJECT_NAME}
+      ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+      LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+      RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+    )
 
-#############
-## Testing ##
-#############
-if (CATKIN_ENABLE_TESTING)
-catkin_add_gtest(${PROJECT_NAME}-test
-                 test/gtest_main.cc
-                 test/composite_test.cc
-                 test/problem_test.cc)
-if(TARGET ${PROJECT_NAME}-test)
-  target_link_libraries(${PROJECT_NAME}-test 
-    ${PROJECT_NAME}
-    pthread
-  )
-endif()
```

Comment:
It would be great if the code for cmake build wouldn't have to be repeated for catkin build. But i'm not sure how to handle this in the most flexible way.